clients/drcachesim/tracer/raw2trace.h

```diff
@@ -388,6 +391,22 @@ struct trace_metadata_reader_t {
     check_entry_thread_start(const offline_entry_t *entry);
 };
 
+#ifdef BUILD_PT_POST_PROCESSOR
+/**
+ * Functions for decoding and verifying raw memtrace data headers.
+ */
+struct ktrace_metadata_reader_t {
+    static bool
+    is_pt_metadata_header(const syscall_pt_entry_t *entry);
```

Comment:
style: A struct isn't supposed to have functions; if it has more than constructors we generally make it a class.

Reply:
These functions are very simple and only be called in a single location. Therefore, it would be most efficient to inline all of these functions.

Reply:
IIRC you pointed out to me that we also have the similar trace_metadata_reader_t which is also a struct with functions.

Reply:
And that we wanted to do the same thing for kernel traces to allow for more complex logic in future.

Reply:
Looking -- ok I see this trace_metadata_reader_t and _writer_t.  They don't seem appealing to me.  I don't think we should model new code on those.  Probably they were part of the raw2trace refactoring done by someone else who was trying to move pieces to external code for parallelism -- but we don't need that anymore.